### PR TITLE
FEAT(billing_service): implemented API and tests

### PR DESCRIPTION
**Pull request by jniles**
This commit completely implements the server-side API and integration tests for Billing Services.

### CODE REVIEW

File: server/config/routes.js [hunk 1]
```diff
@@ -378,20 +379,27 @@ exports.configure = function (app) {
...
-  // employees api
+  /** employees */
   app.get('/employees', employees.list);
   app.get('/employees/:id', employees.details);
   app.put('/employees/:id', employees.update);
   app.post('/employees', employees.create);
+
+  /** billing services */
+  app.get('/billing_services', billingServices.list);
+  app.get('/billing_services/:id', billingServices.details);
```
Comment (DedrickEnc) [inline server/config/routes.js]: Let use the verb to express an interface, `getDetail`` is better.

Comment (sfount) [inline server/config/routes.js]: The verb is in the HTTP request! 

In this case the verb might be **to detail** vs **details**. If we were going to strictly follow the **create**, **update**, **list** standards the verb here would be `billingServices.detail`. Whatever we decide we should use it uniformly everywhere.

Comment (jniles) [inline server/config/routes.js]: Alright, I'll change the controller to read **detail** as for the GET `/entities/:id` request.

**Comment by jniles** [comment]
@sfount, this can be re-reviewed anytime.  The suggested changes have been implemented.

**Comment by sfount** [comment]
@jniles waiting on the latest MySQL to be able to review these database changes on. I'll review this as soon as this is available.

**Pull request merged**